python/sglang/srt/models/torch_native_llama.py

```diff
@@ -495,8 +554,43 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
             param = self.lm_head.weight
             weight_loader = getattr(param, "weight_loader", default_weight_loader)
             weight_loader(param, self.model.embed_tokens.weight)
+
+        # Re-arrange fused matrix for TP
+        tp_size = get_tensor_model_parallel_world_size()
```

Comment:
here, is it possible to do:

1. split qkv to 3 Tensors
2. apply tp to each of the Tensor
3. concat 3 Tensors to a single DTensor

This way we can rely on split/concat ops in DTensor itself instead of worrying about the implementation details?

Reply:
Maybe? Although, at this location, we haven't applied TP yet, so there is no notion of DTensor yet.

Reply:
Hmm, I see what you mean. We can use DTensor API instead of TP API (higher level) here.

Reply:
In the newer version, I added support for TP'lized weight loading. Then we directly construct DTensor from the local shard. See the `ColwiseParallelSharded` strategy.

Reply:
just to understand, currently step 2 is manual right?

Reply:
Not manual per se. It is already packaged and can be called with `parallelize_module` as like other styles. So no evolvement needed from user.